### PR TITLE
Vlad_Bilanchuk_homework_1

### PR DESCRIPTION
**Pull request by LockiStrike**

### CODE REVIEW

**Review by oleksandr-kychun — COMMENTED** [review]
Interesting solution) how about "magic ranges" and "magic  numbers"? any idea how to get rid of them?

File: homework.rb [hunk 1]
```diff
@@ -1,6 +1,17 @@
...
+  result = ''
+  2.times { result += rand(65..90).chr}
+  result += '-'
+  result += rand(1000000..9999999).to_s
+  result += '-'
+  3.times { result += rand(65..90).chr}
+  result
+
+  #solution in one line but slower in 100 times
+  # ('AA'..'ZZ').to_a.sample + '-' + rand(1000000..9999999).to_s + '-' + ('AAA'..'ZZZ').to_a.sample
```
Comment (sveredyuk) [inline homework.rb]: what about something like:
```ruby
# define some tmp variables
"#{one_part}-#{second_part}-#{third_part]"
```

File: homework.rb [hunk 2]
```diff
@@ -1,6 +1,17 @@
...
+VARIANT = 5
 
 def generate
-  # put your solution here (delete the line below)
-  rand(1000000).to_s + '-' + rand(1000000).to_s
-end
+
+  # fastest finishing test
+  result = ''
+  2.times { result += rand(65..90).chr}
```
Comment (sergiy-koshoviy) [inline homework.rb]: 2, 3 - magic numbers

**Comment by LockiStrike** [comment]
tried to remove magic numbers by adding pattern parser

File: homework.rb [hunk 3]
```diff
@@ -1,6 +1,64 @@
-VARIANT = 0
+VARIANT = 5
+
+LETTER_START_POINT = 65
+LETTER_END_POINT = 90
+# pattern = 'AC-1234567-ABC'
+
+def sequence_nature(pattern='AC-1234567-ABC', index=0)
```
Comment (ruslan-369) [inline homework.rb]: move pattern default value to constant

File: homework.rb [hunk 4]
```diff
@@ -1,6 +1,64 @@
...
+  if pattern[index].match /[a-z]|[A-Z]/ # pattern[/[a-z]|[A-Z]/]
+    'letter'
+  elsif pattern[index].match /[0-9]/
+    'number'
+  else
+    'error'
+  end
```
Comment (ruslan-369) [inline homework.rb]: better to use case statement here

Comment (ruslan-369) [reply homework.rb]: move regexp patterns to constants

File: homework.rb (hunk 3 above)
Comment (ruslan-369) [reply homework.rb]: if u are using a single element of pattern here, is there need to pass whole pattern with needed index?

**Comment by ruslan-369** [comment]
Why did u rewrite code at all after receiving comments from other guys? The task is about implementing randomizer for YOUR preset pattern instead of the general solution for all patterns... As for me it is similar to situation if your client asks you to implement simple blog but you are implementing a kind of blog builder)) Also do u understand how your code works?)

**Comment by LockiStrike** [comment]
I thought that write some constants like FIRST_PART_COUNT or smth like this is weird and not what you want. I agree that last commit is not about randomizer for my preset pattern. Of course i understand how my code works, there are many bad places. I'll change it to previous variant with your comments.

File: homework.rb [hunk 5]
```diff
@@ -1,6 +1,64 @@
...
+
+    return pattern_info
+  else
+    'error'
+  end
+
+end
+
+def pattern_randomizer(pattern='AC-1234567-ABC')
+  pattern_info = pattern_parser(pattern)
```
Comment (sveredyuk) [inline homework.rb]: over engineering

File: homework.rb [hunk 6]
```diff
@@ -14,15 +13,13 @@
...
-  
-  second_part = ''
-  SECOND_PART_LENGTH.times { second_part += rand(DIGITS_RANGE).to_s}
-  
-  third_part = ''
-  THIRD_PART_LENGTH.times { third_part += rand(UPCASE_LETTERS_RANGE).chr}
-  result = "#{first_part}-#{second_part}-#{third_part}"
+
+
+  first_part = ->()  { (DIGITS_RANGE).to_a.sample(FIRST_PART_LENGTH).join }
```
Comment (sveredyuk) [inline homework.rb]: if lamda has no args:
`first_part = -> { (DIGITS_RANGE).to_a.sample(FIRST_PART_LENGTH).join }`

**Pull request closed without merging**